src/result_object/index.js

```diff
@@ -62,10 +61,29 @@ class ResultObject {
      * @param {string} fieldName - The name of the field.
      * @param {string} statement - The error string to add to the object.
      * @param {string} severity - Whether it is a `fail` or `warn` test.
-     * @return {Function} Calling a helper function
+     * @return {Object} Current instance
      */
-    fail(fieldName: string, statement: string, severity: Severity = FAIL) {
-        return fail.call(this, fieldName, statement, severity);
+    fail(fieldName: string, statement: string, severity: Severity) {
+        const selectedSeverity: Severity = severity && severities.includes(severity) ? severity : FAIL;
+        const isWarn: boolean = selectedSeverity === WARN;
+
+        const countName: string = isWarn ? 'warnCount' : 'failCount';
+
+        if (isWarn) {
+            this.hasValidationWarnings = true;
+            this.validationWarnings[fieldName] = this.validationWarnings[fieldName] || [];
+            this.validationWarnings[fieldName].push(statement);
+            this.warnCount++;
+        } else {
+            this.hasValidationErrors = true;
+            this.validationErrors[fieldName] = this.validationErrors[fieldName] || [];
+            this.validationErrors[fieldName].push(statement);
+            this.failCount++;
+        }
+
+        this.testsPerformed[fieldName][countName]++;
+
+        return this;
```

Comment:
I don't like the functions `getErrors`, do you think it's possible to add a getter to `this.errors` and another getter (more complicated) on `this[fieldName].errors`. Same for warnings, I think it's very nice for the user to simply do `res[fieldName].errors` or `res[fieldName].warnings`. I don't think you need the nested `validationErrors` and `validationWarnings`.

Reply:
The question is how to do that?
Maybe you can use `Proxy`? something like
```js
new Proxy(passable.res, {
    get: function(target, name, receiver) {
        switch(name) {
            case 'errors':
                return this.getErrors(target);
            case 'warnings':
                return this.getWarnings(target);
            default:
                return target[name];
        }
     }
});
```
outside the class, when creating the passable instance.

Reply:
no no, not a good idea it won't work at all

Reply:
azov

Reply:
>I don't like the functions getErrors, do you think it's possible to add a getter to this.errors and another getter (more complicated) on this[fieldName].errors. Same for warnings, I think it's very nice for the user to simply do res[fieldName].errors or res[fieldName].warnings. I don't think you need the nested validationErrors and validationWarnings.

I actually think it simplifies working with passable.
Think about cases that I have multiple fields with errors.

```js
const result = passable(...);
```

now, to display the errors, I would need to do something like this:

```js
const usernameErr = res.validationErrors.username.length && res.validationErrors.username[0];
const passwordErr = res.validationErrors.password.length && res.validationErrors.password[0];
const emailErr = res.validationErrors.email.length && res.validationErrors.email[0];
```

This is basically a shortcut that does the same thing.